orttraining/orttraining/training_ops/cuda/optimizer/lamb.cc

```diff
@@ -267,6 +267,7 @@ Status launch_lamb_compute_direction(
 
 template <typename CudaTNorm, typename CudaTIn1, typename CudaTIn2>
 Status launch_lamb_reduction(
+    const CudaKernel* kernel,
```

Comment:
>const CudaKernel* [](start = 4, length = 17)

can this be `const CudaKernel&` so it's not possible to have a null value?

Reply:
Well, it's an upcast from LambOptimizer* to CudaKernel*.

Reply:
that should work for references too

---
In reply to: [539740511](https://github.com/microsoft/onnxruntime/pull/6028#discussion_r539740511) [](ancestors = 539740511)

Reply:
Ah, OK. I didn't realize that.

I gave it a try converting '*' to '&' in signatures but got an error I'm unsure on
```
/nfs/sukha/onnxruntime/onnxruntime/core/providers/cuda/multi_tensor/common.cuh:75:6: error: 
invalid abstract parameter type ‘onnxruntime::cuda::CudaKernel’
```

Reply:
Resolved using perfect forwarding. Thanks for the help!